shared/js/background/chrome-events.es6.js

```diff
@@ -75,16 +76,13 @@ chrome.webRequest.onHeadersReceived.addListener(
                 if (!request.responseHeaders) return
                 if (tab && tab.site.whitelisted) return
                 if (tab && utils.isFirstParty(request.url, tab.url)) return
-                const index = request.responseHeaders.findIndex(header => { return header.name.toLowerCase() === 'set-cookie' })
-                if (index !== -1) {
-                    if (!cookieConfig.isExcluded(request.url) && trackerutils.isTracker(request.url)) {
-                        request.responseHeaders.splice(index, 1)
-                    }
+                if (!cookieConfig.isExcluded(request.url) && trackerutils.isTracker(request.url)) {
+                    responseHeaders = responseHeaders.filter(header => header.name.toLowerCase() !== 'set-cookie')
                 }
             }
         }
 
-        return { responseHeaders: request.responseHeaders }
+        return { responseHeaders: responseHeaders }
```

Comment:
```suggestion
        return { responseHeaders }
```

Reply:
I would also consider only attaching `responseHeaders` to the `BlockingResponse` if we actually change the headers. This will prevent any potential overhead for unchanged requests.